src/main/java/com/github/manolo8/darkbot/gui/titlebar/DiagnosticBar.java

```diff
@@ -0,0 +1,106 @@
+package com.github.manolo8.darkbot.gui.titlebar;
+
+import com.formdev.flatlaf.FlatClientProperties;
+import com.formdev.flatlaf.extras.FlatSVGIcon;
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.core.api.Capability;
+import com.github.manolo8.darkbot.core.manager.StatsManager;
+import com.github.manolo8.darkbot.gui.components.DiagnosticsPanel;
+import com.github.manolo8.darkbot.gui.utils.UIUtils;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.MouseMotionListener;
+import java.util.function.Function;
+
+public class DiagnosticBar extends JButton {
+
+    DiagnosticBar(Main main) {
+        StatsManager statsManager = main.statsManager;
+
+        setBorder(BorderFactory.createEmptyBorder());
+        setLayout(new MigLayout("ins 0, gap 0", "3px:5px[][right]3px:5px", "[15px!][15px!]"));
+        putClientProperty(FlatClientProperties.BUTTON_TYPE, FlatClientProperties.BUTTON_TYPE_BORDERLESS);
+
+        JLabel tick = createLabel("tick_time", "Tick time", false,
+                color -> UIUtils.getTrafficLight(main.getTickTime(), 30), statsManager.getTickStats());
+        JLabel ping = createLabel("ping", "In-game ping", false,
+                color -> UIUtils.getTrafficLight(main.statsManager.getPing(), 300), statsManager.getPingStats());
+        add(tick, "cell 0 0");
+        add(ping, "cell 0 1");
+
+        if (Main.API.hasCapability(Capability.HANDLER_CPU_USAGE, Capability.HANDLER_RAM_USAGE)) {
+            JLabel cpu = createLabel("cpu", "Cpu usage", true,
+                    color -> UIUtils.getTrafficLight(Main.API.getCpuUsage(), 100), statsManager.getCpuStats());
+            JLabel ram = createLabel("ram", "Ram usage", true,
+                    color -> UIUtils.getTrafficLight(Main.API.getMemoryUsage(), 2500), statsManager.getMemoryStats());
+            add(cpu, "cell 1 0, gapleft 5px");
+            add(ram, "cell 1 1, gapleft 5px");
+        }
+
+        for (Component component : getComponents()) {
+            component.addMouseListener(new RedirectMouseAdapter());
+            component.addMouseMotionListener(new RedirectMouseAdapter());
```

Comment:
try to use a layered ui for the parent to capture all events occurring inside its bounds, instead of attaching redirecting listeners to each child